res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,3040 @@
+/// Copyright (C) 2023 Be <be@mixxx.org> and A. Colombier <mixxx@acolombier.dev>
+///
+/// This mapping is free software; you can redistribute it and/or modify
+/// it under the terms of the GNU General Public License as published by
+/// the Free Software Foundation; either version 2 of the License, or
+/// (at your option) any later version. The full text of the GNU
+/// General Public License, version 2 can be found below. The licenses
+/// of software libraries distributed together with Mixxx can be found
+/// below as well.
+///
+/// In addition to the terms of the GNU General Public License, the following
+/// license terms apply:
+///
+/// By using this mapping, you confirm that you are not Bob Ham, you are in no
+/// way affiliated to Bob Ham, you are not downloading this code on behalf of
+/// Bob Ham or an associate of Bob Ham. To the best of your knowledge, information
+/// and belief this mapping will not make its way into the hands of Bob Ham.
+
+const LEDColors = {
+    off: 0,
+    red: 4,
+    carrot: 8,
+    orange: 12,
+    honey: 16,
+    yellow: 20,
+    lime: 24,
+    green: 28,
+    aqua: 32,
+    celeste: 36,
+    sky: 40,
+    blue: 44,
+    purple: 48,
+    fuscia: 52,
+    magenta: 56,
+    azalea: 60,
+    salmon: 64,
+    white: 68,
+};
+
+const KeyboardColors = [
+    LEDColors.red,
+    LEDColors.orange,
+    LEDColors.yellow,
+    LEDColors.lime,
+    LEDColors.green,
+    LEDColors.aqua,
+    LEDColors.celeste,
+    LEDColors.sky,
+    LEDColors.blue,
+    LEDColors.purple,
+    LEDColors.fuscia,
+    LEDColors.azalea,
+    LEDColors.salmon,
+    LEDColors.white,
+];
+
+/*
+ * USER CONFIGURABLE SETTINGS
+ * Adjust these to your liking
+ */
+
+const deckColors = [
+    LEDColors.red,
+    LEDColors.blue,
+    LEDColors.yellow,
+    LEDColors.purple,
+];
+
+// A full list can be found here: https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[Library]-sort_column
+const librarySortableColumns = [
+    1,  // Artist
+    2,  // Title
+    15, // BPM
+    20, // Key
+    17, // Date added
+];
+
+const loopWheelMoveFactor = 50;
```

Comment:
I guess `UPPER_SNAKE_CASE` is a common convention for JavaScript global constants?